packages/app/src/specs/SpecsListBanners.vue

```diff
@@ -272,6 +276,46 @@ watch(
   { immediate: true },
 )
 
+const bannerCohortOptions = {
+  [BannerIds.ACI_082022_LOGIN]: [
+    { cohort: 'A', value: t('specPage.banners.login.contentA') },
+    { cohort: 'B', value: t('specPage.banners.login.contentB') },
+  ],
+  [BannerIds.ACI_082022_CREATE_ORG]: [
+    { cohort: 'A', value: t('specPage.banners.createOrganization.titleA') },
+    { cohort: 'B', value: t('specPage.banners.createOrganization.titleB') },
+  ],
+  [BannerIds.ACI_082022_CONNECT_PROJECT]: [
+    { cohort: 'A', value: t('specPage.banners.connectProject.contentA') },
+    { cohort: 'B', value: t('specPage.banners.connectProject.contentB') },
+  ],
+}
+
+const getCohortForBanner = (bannerId: string) => {
+  const cohortConfig: CohortConfig = {
+    name: bannerId,
+    options: bannerCohortOptions[bannerId],
+  }
+
+  return useCohorts(cohortConfig)
+}
+
+const cohorts = reactive({} as Record<'login' | 'connectProject' | 'organization', any>)
```

Comment:
I have some questions about this (sorry I only just noticed as part of the re-review...)

This type and usage felt a little off. I found out why - we have a kind of "double reactivity" here. We are using `reactive` to hold properties which all point to `Ref` (which is also part of the reactivity system).

It's similar to doing something like

```js
const [foo1, setFoo1] = useState()
const [foo1, setFoo1] = useState()

const [foos, setFoos] = useState()
setFoos({ setFoo1, setFoo2 })
```

I'm not sure if this really makes sense. I tried a few things, and landed on https://github.com/cypress-io/cypress/compare/stokes/23083_aci_cohorts...lmiller/stokes/23083_aci_cohorts-v1?expand=1.

```ts
type BannerType = 'login' | 'connectProject' | 'organization'
const cohorts: Partial<Record<BannerType, ReturnType<typeof getCohortForBanner>>> = {}
```

Alternatively, just have 3 separate variables:

```ts
const cohortLogin = ref<ReturnType<typeof getCohortForBanner> | undefined>()
// ...
```

This might be more simple - avoid thinking about the reactivity problem all together.
\
I don't know if `cohorts` actually needs to be `reactive`. Either way, this change revealed a bug - `getCohortForBanner` which returns `useCohorts` which returns `ref<CohortOption | undefined>()`. I got a TS error, which I fixed by doing an additional null check (see above link). 

The properties (`cohorts.login` etc) are nullable - but we are passing them to `<ConnectProjectBanner />` etc, which has a non-nullable prop type. I think we need this null check.

We missed this because of the `as Record<any>` cast. Can a user be in *no* cohort? If they are, I don't know if we'd want to display the modals.

Not sure if this actual manifested in a functional bug, but definitely worth clarifying + improving the types (ideally avoiding `any`).

Reply:
I do not believe there is a double reactivity problem because Vue unwraps refs in side `reactive`. See: https://vuejs.org/guide/essentials/reactivity-fundamentals.html#ref-unwrapping-in-reactive-objects

Also, I do still need the cohorts to be reactive because of the mutation it uses is async.  I was following this pattern for my composable: https://www.vuemastery.com/blog/coding-better-composables-5-of-5/

The problem with using `reactive` and trying to strongly type the value part of the `Record` is the unwrapping the `reactive` does.  You can assign either a `ref` or the unwrapped value any place inside a `reactive` object, but would always access it as unwrapped.  

I liked using `reactive` because of the unwrapping so you would not have to do all the `.value` calls in the template.  I thought about the separate variables as well, but they would have to be double nested `ref`s or just be defined with `let` and set after the `getCohortForBanner` returns.  

I am checking in an update that looks pretty ugly, but does seem to work that defines the `cohorts` variable like:
```
type BannerType = 'login' | 'connectProject' | 'organization'

const cohorts: Partial<Record<BannerType, CohortOption | Ref<CohortOption | undefined>>> = reactive({})
```

Let me know what you think.

Reply:
Nevermind.... Went back to the `any` for the moment since that did introduce another type check issue.

Reply:
Right I see why `cohorts` needs to be `reactive` now - the refs are initially `undefined` and (possibly) assigned later on.

I think the original problem still stands doesn't it? By using `any` are we missing out on some value typescript static checking?


I won't hold this one up too much more - I could be missing something, I thought this would be straight forward to type, but maybe that's not the case.

Reply:
Ended up going with your answer but just left off the last part of the null check since the CohortOption, if present, will always have a cohort